---
title: "Myositis-Related Intersitial Lung disease at a single cell level"
author: Serdar Korur
date: '2019-07-15'
slug: scrnaseq
categories: []
tags:
  - Clustering
  - pca
  - tsne
  - seurat
subtitle: ''
summary: ''
authors: []
lastmod: '2019-07-15T16:41:47+02:00'
featured: no
image:
  caption: ''
  focal_point: ''
  preview_only: no
projects: []
---



<div id="lets-explore-gene-expression-and-cellular-subtypes-in-fibrosis-hints-from-single-cell-level" class="section level1">
<h1>Let’s explore gene expression and cellular subtypes in fibrosis: Hints from single cell level</h1>
<div id="what-can-we-learn-from-a-myositis-related-intersitial-lung-disease-patient-data" class="section level2">
<h2>“What can we learn from a Myositis-Related Intersitial Lung disease patient <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM3489196">data?</a>”</h2>
<p>I used the data from a recent <a href="https://www.atsjournals.org/doi/pdf/10.1164/rccm.201712-2410OC">paper</a> to explore genomic landscape in a particular patient with Pulmonary fibrosis.</p>
<p>To make my own analyses I picked up the data from a patient with PM-ILD.</p>
<p>—&gt; <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM3489196">Download Single Cell RNA seq data</a></p>
</div>
<div id="preparations" class="section level2">
<h2>Preparations</h2>
<div id="load-the-necessary-packages" class="section level3">
<h3>Load the necessary packages</h3>
<pre class="r"><code>library(Seurat)
library(dplyr)
library(ggplot2)</code></pre>
</div>
<div id="read-the-data-with-read10x_h5-function-and-create-the-seurat-object" class="section level3">
<h3>Read the Data with <strong>Read10X_h5</strong> function and Create the Seurat object</h3>
<pre class="r"><code>MyoILD_01 &lt;- Read10X_h5(&quot;GSM3489196_Myositis-ILD_01_filtered_gene_bc_matrices_h5.h5&quot;)
Fib &lt;- CreateSeuratObject(counts = MyoILD_01, min.cells = 3, project= &quot;FightFibrosis&quot;, min.features = 200)
Fib</code></pre>
<pre><code>## An object of class Seurat 
## 20246 features across 7163 samples within 1 assay 
## Active assay: RNA (20246 features)</code></pre>
<pre class="r"><code>a &lt;- dim(Fib)</code></pre>
<p>Our Data matrix now contains 20246 genes and 7163 cells</p>
</div>
</div>
<div id="lets-have-a-quick-look-at-the-raw-count-data" class="section level2">
<h2>Let’s have a Quick look at the raw count data</h2>
<pre class="r"><code># Lets examine a few genes in the first thirty cells
MyoILD_01[c(&quot;SPP1&quot;,&quot;TGFB1&quot;,&quot;CCL2&quot;), 1:30]</code></pre>
<pre><code>## 3 x 30 sparse Matrix of class &quot;dgCMatrix&quot;
##                                                                    
## SPP1  1 . . 2 . . . 33 . . 1 . . . . . . . . . . . . . . . 9 . . 20
## TGFB1 . . . . 1 . .  . . . . . . . 1 . 3 . . . . . . . 1 2 . . .  .
## CCL2  5 . . 2 . . .  . . . . . . . . . . . . . . . . . . . . . .  .</code></pre>
<div id="standard-pre-processing-workflow-for-scrna-seq-data" class="section level3">
<h3>Standard pre-processing workflow for scRNA-seq data</h3>
<p>Pre-processing involves filtration of cells based on Quality control metrics (e.g .mitochondrial contamination, Coverage). We will proceed by normalization and identification of the highly variable features then at the end we will scale the data.</p>
<div id="qc-and-selecting-cells-for-further-analysis" class="section level4">
<h4>QC and selecting cells for further analysis</h4>
<pre class="r"><code># Let&#39;s add Mitochondrial stats to Fib data. [[ operator can add columns to object metadata. 

Fib[[&quot;percent.mt&quot;]] &lt;- PercentageFeatureSet(object = Fib, pattern = &quot;^MT-&quot;)</code></pre>
<pre class="r"><code># Quick look at the Quality control metrics for the first 5 cells
head(x = Fib@meta.data, 5)</code></pre>
<pre><code>##                       orig.ident nCount_RNA nFeature_RNA percent.mt
## AAACCTGAGATCCCGC-1 FightFibrosis       2257         1010   4.386354
## AAACCTGAGCAGCCTC-1 FightFibrosis       3367          713  66.914167
## AAACCTGAGCCAGAAC-1 FightFibrosis       7354         1943   3.929834
## AAACCTGAGGGATGGG-1 FightFibrosis       9323         2786   8.430763
## AAACCTGAGTCGTACT-1 FightFibrosis       2937         1125   6.775621</code></pre>
</div>
<div id="visualize-qc-metrics" class="section level4">
<h4>Visualize QC metrics</h4>
<pre class="r"><code>#Visualize QC metrics as a violin plot
VlnPlot(object = Fib, features = c(&quot;nFeature_RNA&quot;, &quot;nCount_RNA&quot;, &quot;percent.mt&quot;), ncol = 3)</code></pre>
<p><img src="/post/2019-07-15-scrnaseq_files/figure-html/qc2-1.png" width="1248" /></p>
<pre class="r"><code># We will use FeatureScatter to visualize feature-feature relationships, but can be used for anything calculated  by the object, i.e. columns in object metadata, PC scores etc.
plot1 &lt;- FeatureScatter(object = Fib, feature1 = &quot;nCount_RNA&quot;, feature2 = &quot;percent.mt&quot;) 
plot2 &lt;- FeatureScatter(object = Fib, feature1 = &quot;nCount_RNA&quot;, feature2 = &quot;nFeature_RNA&quot;) 
CombinePlots(plots = list(plot1,plot2))</code></pre>
<p><img src="/post/2019-07-15-scrnaseq_files/figure-html/qc2-2.png" width="1248" /></p>
<pre class="r"><code># As you see on the left plot cells with high percentage of mitochondrial genes have very low numbers of RNA indicating that they are low quality/dead cells. Let&#39;s remove them.
Fib &lt;- subset(x = Fib, subset = nFeature_RNA &gt; 200 &amp; nFeature_RNA &lt; 4000 &amp; percent.mt &lt; 12.5)</code></pre>
</div>
<div id="normalize-the-data" class="section level4">
<h4>Normalize the data</h4>
<p>Log normalization helps to reduce the influences of the outliers. Normalized values will be stored in <code>Fib[["RNA"]]@data</code>.</p>
<pre class="r"><code>Fib &lt;- NormalizeData(object = Fib, normalization.method = &quot;LogNormalize&quot;, scale.factor = 1e4)</code></pre>
</div>
<div id="identify-highly-variable-features" class="section level4">
<h4>Identify highly variable features</h4>
<p>Let’s calculate the features that exhibit high cell-to-cell variation in the dataset. Focusing on those genes will help to highlight biological signal.</p>
<pre class="r"><code>Fib &lt;- FindVariableFeatures(object = Fib, selection.method = &#39;vst&#39;, nfeatures = 2000)

# Identify the 10 most highly variable genes
top10 &lt;- head(x = VariableFeatures(object = Fib), 10)
top10</code></pre>
<pre><code>##  [1] &quot;SCGB1A1&quot; &quot;SCGB3A1&quot; &quot;SFTPC&quot;   &quot;SCGB3A2&quot; &quot;HBB&quot;     &quot;PLA2G2A&quot; &quot;MT1G&quot;   
##  [8] &quot;TPSB2&quot;   &quot;FABP4&quot;   &quot;MT1H&quot;</code></pre>
<pre class="r"><code># Plot variable features with and without labels
plot1 &lt;- VariableFeaturePlot(object = Fib)
plot2 &lt;- LabelPoints(plot = plot1, points = top10, repel = TRUE)
CombinePlots(plots = list(plot1, plot2))</code></pre>
<p><img src="/post/2019-07-15-scrnaseq_files/figure-html/var_features-1.png" width="1056" /></p>
</div>
<div id="scale-the-data" class="section level4">
<h4>Scale the data</h4>
<p>Next, apply a linear transformation that is a standard pre-processing step prior to dimensional reduction techniques like PCA. The <code>ScaleData</code> function:</p>
<ul>
<li>Shifts the expression of each gene, so that the mean expression across cells is 0</li>
<li>Scales the expression of each gene, so that the variance across cells is 1
<ul>
<li><strong>This step gives equal weight in downstream analyses, so that highly-expressed genes do not dominate</strong>
Any systematic difference in count size across the non-DE majority of genes between two cells is assumed to represent bias and is removed by scaling.</li>
</ul></li>
<li>The results of this are stored in <code>Fib[["RNA"]]@scale.data</code></li>
</ul>
<pre class="r"><code>all.genes &lt;- rownames(x = Fib)
Fib &lt;- ScaleData(object = Fib, features = all.genes)</code></pre>
<pre class="r"><code># A list of cell cycle markers, from Tirosh et al, 2015, is loaded with Seurat.  We can
# segregate this list into markers of G2/M phase and markers of S phase
s.genes &lt;- cc.genes$s.genes
g2m.genes &lt;- cc.genes$g2m.genes
Fib &lt;- CellCycleScoring(Fib, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)

# view cell cycle scores and phase assignments
head(Fib[[]])</code></pre>
<pre><code>##                       orig.ident nCount_RNA nFeature_RNA percent.mt
## AAACCTGAGATCCCGC-1 FightFibrosis       2257         1010   4.386354
## AAACCTGAGCCAGAAC-1 FightFibrosis       7354         1943   3.929834
## AAACCTGAGGGATGGG-1 FightFibrosis       9323         2786   8.430763
## AAACCTGAGTCGTACT-1 FightFibrosis       2937         1125   6.775621
## AAACCTGAGTCGTTTG-1 FightFibrosis       5447         1719   4.149073
## AAACCTGCAAGTCTAC-1 FightFibrosis       6453         1836   2.773904
##                         S.Score    G2M.Score Phase     old.ident
## AAACCTGAGATCCCGC-1  0.006739441 -0.016659228     S FightFibrosis
## AAACCTGAGCCAGAAC-1 -0.040762255 -0.010151464    G1 FightFibrosis
## AAACCTGAGGGATGGG-1 -0.048223904 -0.025580453    G1 FightFibrosis
## AAACCTGAGTCGTACT-1  0.076114887  0.009717348     S FightFibrosis
## AAACCTGAGTCGTTTG-1 -0.045171005 -0.053868788    G1 FightFibrosis
## AAACCTGCAAGTCTAC-1  0.003354484 -0.009033119     S FightFibrosis</code></pre>
<pre class="r"><code># Regress out cell cycle scores during data scaling
Fib &lt;- ScaleData(Fib, vars.to.regress = c(&quot;S.Score&quot;, &quot;G2M.Score&quot;), features = rownames(Fib))</code></pre>
</div>
</div>
<div id="perform-linear-dimensional-reduction" class="section level3">
<h3>Perform linear dimensional reduction</h3>
<p>Next we perform PCA on the scaled data.</p>
<pre class="r"><code>Fib &lt;- RunPCA(object = Fib, features = VariableFeatures(object = Fib), ndims.print = 10, nfeatures.print = 10)</code></pre>
<div id="visualize-the-pca-results" class="section level4">
<h4>Visualize the PCA results</h4>
<pre class="r"><code># Examine and visualize PCA results a few different ways
print(x = Fib[[&#39;pca&#39;]], dims = 1:5, nfeatures = 5)</code></pre>
<pre><code>## PC_ 1 
## Positive:  TMSB4X, TYROBP, FTL, FCER1G, VIM 
## Negative:  RSPH1, C9orf24, TMEM190, C20orf85, FAM183A 
## PC_ 2 
## Positive:  FTH1, CTSS, TYROBP, B2M, FCER1G 
## Negative:  GPRC5A, SFTPB, CEACAM6, HOPX, MUC1 
## PC_ 3 
## Positive:  PLA2G2A, GFPT2, MEDAG, COL3A1, COL1A2 
## Negative:  SFTPB, SFTPA2, SFTA2, SFTPA1, SLC34A2 
## PC_ 4 
## Positive:  SFTPC, NAPSA, SFTPD, LRRK2, LAMP3 
## Negative:  S100A2, KRT19, KRT5, AKR1C1, BPIFB1 
## PC_ 5 
## Positive:  CPA3, TPSAB1, TPSB2, MS4A2, GATA2 
## Negative:  S100A6, CSTB, S100A10, GCHFR, TXN</code></pre>
<pre class="r"><code>VizDimLoadings(object = Fib, dims = 1:2, reduction = &#39;pca&#39;)</code></pre>
<p><img src="/post/2019-07-15-scrnaseq_files/figure-html/pca_viz-1.png" width="672" /></p>
<pre class="r"><code>DimPlot(object = Fib, reduction = &#39;pca&#39;)</code></pre>
<p><img src="/post/2019-07-15-scrnaseq_files/figure-html/pca_viz-2.png" width="672" /></p>
<pre class="r"><code>DimHeatmap(object = Fib, dims = 1, cells = 500, balanced = TRUE)</code></pre>
<p><img src="/post/2019-07-15-scrnaseq_files/figure-html/multi-heatmap-1.png" width="864" /></p>
<pre class="r"><code>DimHeatmap(object = Fib, dims = 1:9, cells = 500, balanced = TRUE)</code></pre>
<p><img src="/post/2019-07-15-scrnaseq_files/figure-html/multi-heatmap-2.png" width="864" /></p>
</div>
</div>
<div id="determine-the-dimensionality-of-the-dataset" class="section level3">
<h3>Determine the ‘dimensionality’ of the dataset</h3>
<p>To overcome the extensive technical noise in any single feature for scRNA-seq data, Seurat clusters cells based on their PCA scores, with each PC essentially representing a ‘metafeature’ that combines information across a correlated feature set. The top principal components therefore represent a robust compression of the dataset.</p>
<p>We will ‘Elbow plot’ to determine the dimensionalitz of this dataset: a ranking of principle components based on the percentage of variance explained by each one (<code>ElbowPlot</code> function). In this example, we can observe an ‘elbow’ around PC9-10, suggesting that the majority of true signal is captured in the first 10 PCs.</p>
<pre class="r"><code>ElbowPlot(object = Fib)</code></pre>
<p><img src="/post/2019-07-15-scrnaseq_files/figure-html/elbow_plot-1.png" width="960" /></p>
</div>
<div id="cluster-the-cells" class="section level3">
<h3>Cluster the cells</h3>
<p>Seurat v3 applies a graph-based clustering approach, building upon initial strategies in (<a href="http://www.cell.com/abstract/S0092-8674(15)00549-8">Macosko <em>et al</em></a>).</p>
<pre class="r"><code>Fib &lt;- FindNeighbors(object = Fib, dims = 1:10)</code></pre>
<pre><code>## Computing nearest neighbor graph</code></pre>
<pre><code>## Computing SNN</code></pre>
<pre class="r"><code>Fib &lt;- FindClusters(object = Fib, resolution = 0.5)</code></pre>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 6320
## Number of edges: 196713
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.8872
## Number of communities: 15
## Elapsed time: 0 seconds</code></pre>
<pre class="r"><code># Look at cluster IDs of the first 5 cells
head(Idents(Fib), 5)</code></pre>
<pre><code>## AAACCTGAGATCCCGC-1 AAACCTGAGCCAGAAC-1 AAACCTGAGGGATGGG-1 
##                  2                  2                  8 
## AAACCTGAGTCGTACT-1 AAACCTGAGTCGTTTG-1 
##                  0                 10 
## Levels: 0 1 2 3 4 5 6 7 8 9 10 11 12 13 14</code></pre>
</div>
<div id="tsne-plot" class="section level3">
<h3>TSNE plot</h3>
<pre class="r"><code>Fib &lt;- RunTSNE(object = Fib, dims = 1:6)
DimPlot(object = Fib, reduction = &#39;tsne&#39;, label = TRUE, label.size = 5)</code></pre>
<p><img src="/post/2019-07-15-scrnaseq_files/figure-html/TSNE%20plot-1.png" width="672" /></p>
</div>
<div id="finding-differentially-expressed-features-cluster-biomarkers" class="section level3">
<h3>Finding differentially expressed features (cluster biomarkers)</h3>
<p>The <code>min.pct</code> argument requires a feature to be detected at a minimum percentage in either of the two groups of cells.</p>
<pre class="r"><code># Find all markers of cluster 0
cluster0.markers &lt;- FindMarkers(object = Fib, ident.1 = 0, min.pct = 0.25)
head(x = cluster0.markers, n = 10)</code></pre>
<pre><code>##          p_val avg_logFC pct.1 pct.2 p_val_adj
## GPNMB        0 1.0814399 0.947 0.450         0
## CSTB         0 1.0377160 0.994 0.892         0
## FTL          0 1.0160962 1.000 0.998         0
## APOC1        0 0.9919471 0.991 0.819         0
## SH3BGRL3     0 0.9252345 0.990 0.798         0
## CTSD         0 0.9105122 0.995 0.836         0
## FTH1         0 0.8873914 1.000 0.997         0
## PSAP         0 0.8835586 1.000 0.876         0
## CTSB         0 0.8773872 0.986 0.752         0
## LGALS1       0 0.8622147 0.992 0.637         0</code></pre>
<pre class="r"><code># Find markers for every cluster compared to all remaining cells, report only the positive ones
Fib.markers &lt;- FindAllMarkers(object = Fib, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
Fib.markers %&gt;% group_by(cluster) %&gt;% top_n(n = 5, wt = avg_logFC) %&gt;% print(n = 85)</code></pre>
<pre><code>## # A tibble: 75 x 7
## # Groups:   cluster [15]
##        p_val avg_logFC pct.1 pct.2 p_val_adj cluster gene    
##        &lt;dbl&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt; &lt;fct&gt;   &lt;chr&gt;   
##  1 0.            1.08  0.947 0.45  0.        0       GPNMB   
##  2 2.60e-293     1.13  0.935 0.582 5.27e-289 0       FABP5   
##  3 8.54e-289     1.12  0.814 0.365 1.73e-284 0       CTSL    
##  4 3.43e-188     1.43  0.46  0.133 6.95e-184 0       SPP1    
##  5 2.78e-110     1.37  0.288 0.078 5.64e-106 0       CCL2    
##  6 2.69e-231     1.33  0.996 0.658 5.45e-227 1       APOE    
##  7 1.91e-204     1.02  0.926 0.403 3.86e-200 1       MS4A6A  
##  8 1.35e-174     0.918 0.932 0.423 2.74e-170 1       C1QC    
##  9 1.59e-119     1.07  0.86  0.457 3.22e-115 1       CCL18   
## 10 1.56e- 56     0.916 0.576 0.311 3.15e- 52 1       SEPP1   
## 11 0.            2.26  0.663 0.108 0.        2       FABP4   
## 12 0.            1.43  0.846 0.165 0.        2       MCEMP1  
## 13 5.19e-222     1.24  0.969 0.496 1.05e-217 2       GCHFR   
## 14 4.41e-218     1.36  0.98  0.505 8.93e-214 2       C1QA    
## 15 4.43e-210     1.35  0.967 0.483 8.97e-206 2       C1QB    
## 16 0.            3.18  0.964 0.339 0.        3       SCGB3A2 
## 17 0.            2.01  0.995 0.376 0.        3       SFTPB   
## 18 0.            1.73  0.943 0.188 0.        3       CEACAM6 
## 19 2.27e-273     2.23  0.929 0.364 4.59e-269 3       SFTPA2  
## 20 8.38e-271     1.72  0.813 0.203 1.70e-266 3       SFTPA1  
## 21 0.            2.70  0.896 0.035 0.        4       C9orf24 
## 22 0.            2.46  0.87  0.026 0.        4       TMEM190 
## 23 0.            2.44  0.892 0.026 0.        4       RSPH1   
## 24 0.            2.42  0.868 0.025 0.        4       C20orf85
## 25 0.            2.39  0.907 0.104 0.        4       TPPP3   
## 26 0.            2.76  0.975 0.227 0.        5       S100A2  
## 27 0.            2.19  0.797 0.083 0.        5       KRT15   
## 28 0.            1.72  0.684 0.048 0.        5       KRT5    
## 29 6.72e-278     1.93  0.471 0.035 1.36e-273 5       MMP1    
## 30 9.96e-220     1.75  0.925 0.319 2.02e-215 5       AQP3    
## 31 0.            2.45  0.737 0.077 0.        6       BPIFB1  
## 32 6.07e-262     2.14  0.984 0.305 1.23e-257 6       WFDC2   
## 33 2.03e-221     2.21  0.982 0.385 4.11e-217 6       SLPI    
## 34 1.17e-163     3.21  0.948 0.624 2.36e-159 6       SCGB3A1 
## 35 8.07e-152     2.91  0.797 0.293 1.63e-147 6       SCGB1A1 
## 36 0.            2.57  0.797 0.072 0.        7       S100B   
## 37 9.62e-289     1.76  0.829 0.156 1.95e-284 7       FAM26F  
## 38 2.50e-224     2.18  1     0.728 5.06e-220 7       HLA-DPB1
## 39 2.51e-213     1.99  0.992 0.658 5.07e-209 7       HLA-DPA1
## 40 1.75e-191     1.71  0.955 0.449 3.55e-187 7       HLA-DQA1
## 41 4.11e- 50     1.01  0.546 0.231 8.31e- 46 8       STMN1   
## 42 1.29e- 41     0.917 0.819 0.529 2.60e- 37 8       TUBA1B  
## 43 1.29e- 40     0.915 0.807 0.517 2.61e- 36 8       TUBB    
## 44 1.32e- 29     0.973 0.825 0.615 2.67e- 25 8       H2AFZ   
## 45 5.14e- 29     1.16  0.77  0.503 1.04e- 24 8       HIST1H4C
## 46 0.            2.28  0.984 0.113 0.        9       NAPSA   
## 47 3.65e-299     2.20  0.984 0.159 7.39e-295 9       SFTA2   
## 48 8.60e-252     2.99  1     0.226 1.74e-247 9       SFTPA1  
## 49 3.66e-220     4.21  0.961 0.265 7.42e-216 9       SFTPC   
## 50 4.19e-168     2.33  0.992 0.39  8.47e-164 9       SFTPA2  
## 51 0.            2.61  0.570 0.01  0.        10      COL3A1  
## 52 0.            2.56  0.604 0.005 0.        10      SPARCL1 
## 53 0.            2.42  0.456 0.005 0.        10      PLA2G2A 
## 54 0.            2.40  0.544 0.005 0.        10      COL1A2  
## 55 1.09e-278     2.57  0.537 0.02  2.20e-274 10      COL1A1  
## 56 0.            4.23  0.993 0.035 0.        11      TPSB2   
## 57 0.            3.74  0.993 0.02  0.        11      TPSAB1  
## 58 0.            3.32  1     0.02  0.        11      CPA3    
## 59 0.            2.78  0.911 0.017 0.        11      MS4A2   
## 60 0.            2.33  0.856 0.016 0.        11      KIT     
## 61 1.98e-262     2.08  0.624 0.023 4.01e-258 12      FCN1    
## 62 5.17e- 70     1.95  0.475 0.059 1.05e- 65 12      IL1R2   
## 63 2.47e- 66     1.98  0.634 0.119 5.01e- 62 12      VCAN    
## 64 4.37e- 58     3.03  0.812 0.258 8.85e- 54 12      S100A8  
## 65 2.45e- 46     2.67  0.901 0.466 4.96e- 42 12      S100A9  
## 66 2.37e- 68     2.24  0.593 0.055 4.80e- 64 13      IL32    
## 67 1.03e- 66     2.00  0.296 0.013 2.08e- 62 13      KLRB1   
## 68 4.82e- 52     3.79  0.259 0.013 9.77e- 48 13      IGHG1   
## 69 3.98e- 44     3.37  0.259 0.015 8.06e- 40 13      IGHG4   
## 70 3.42e- 42     2.22  0.352 0.03  6.92e- 38 13      CCL5    
## 71 6.32e-208     1.85  0.978 0.036 1.28e-203 14      CPA3    
## 72 1.19e-176     1.28  0.822 0.03  2.41e-172 14      KIT     
## 73 3.11e-176     1.40  0.844 0.032 6.30e-172 14      MS4A2   
## 74 7.63e-172     1.75  0.889 0.036 1.54e-167 14      TPSAB1  
## 75 2.72e-158     2.05  0.978 0.051 5.50e-154 14      TPSB2</code></pre>
<p>Seurat has several tests for differential expression which can be set with the test.use parameter (see our <a href="http://satijalab01.nygenome.org/seurat/v3.0/de_vignette.html">DE vignette</a> for details). For example, the ROC test returns the ‘classification power’ for any individual marker (ranging from 0 - random, to 1 - perfect).</p>
<pre class="r"><code># Assigning Cell ids
new.cluster.ids &lt;- c(&quot;Macrophages(C1)&quot;, &quot;Macrophages(C2)&quot;, &quot;Macrophages(C3)&quot;, &quot;AT2(C1) Cells&quot;, &quot;Cliated Cells&quot;, &quot;Basal Cells&quot;, &quot;Club Cells&quot;, &quot;Dendritic Cells&quot;, &quot;NA&quot;, &quot;AT2(C2) Cells&quot;, &quot;Fibroblasts&quot;, &quot;Mast(T1) Cells&quot;, &quot;Monocytes&quot;, &quot;T/NKT Cells&quot;, &quot;Mast(C2) Cells&quot;)
names(new.cluster.ids) &lt;- levels(Fib)
Fib &lt;- RenameIdents(Fib, new.cluster.ids)
DimPlot(Fib, reduction = &quot;tsne&quot;, label = TRUE, label.size = 6, pt.size = 1.0) + NoLegend()</code></pre>
<p><img src="/post/2019-07-15-scrnaseq_files/figure-html/tsnewithcellids-1.png" width="1440" /></p>
<pre class="r"><code># Let&#39;s look at the epxression of a few genes of interest accross different subtypes
VlnPlot(Fib, features = c(&quot;SPP1&quot;, &quot;CHIT1&quot;))</code></pre>
<p><img src="/post/2019-07-15-scrnaseq_files/figure-html/differentfeautures-1.png" width="672" /></p>
<pre class="r"><code>FeaturePlot(Fib, features = c(&quot;SPP1&quot;, &quot;APOE&quot;, &quot;SCGB3A2&quot;, &quot;C9orf24&quot;, &quot;S100A2&quot;, &quot;BPIFB1&quot;, &quot;S100B&quot;, &quot;COL3A1&quot;, &quot;FCN1&quot;))</code></pre>
<p><img src="/post/2019-07-15-scrnaseq_files/figure-html/differentfeautures-2.png" width="672" /></p>
<pre class="r"><code>top10 &lt;- Fib.markers %&gt;% group_by(cluster) %&gt;% top_n(n = 10, wt = avg_logFC)
DoHeatmap(Fib, features = top10$gene) + NoLegend()</code></pre>
<p><img src="/post/2019-07-15-scrnaseq_files/figure-html/markersroc-1.png" width="672" /></p>
</div>
</div>
</div>
